crates/turborepo-auth/src/auth/login.rs

```diff
@@ -20,13 +30,35 @@ pub async fn login<'a>(
     existing_token: Option<&'a str>,
     login_url_configuration: &str,
     login_server: &impl LoginServer,
-) -> Result<Cow<'a, str>, Error> {
-    // Check if token exists first.
+) -> Result<Token, Error> {
+    // Check if passed in token exists first.
     if let Some(token) = existing_token {
         if let Ok(response) = api_client.get_user(token).await {
             println!("{}", ui.apply(BOLD.apply_to("Existing token found!")));
             ui::print_cli_authorized(&response.user.email, ui);
```

Comment:
This is the chunk of logic that can get shared between the turbo token and vercel token loading.

```
async fn check_token(token: String, api_client: &impl Client, message: &str) -> Result<Token> {
    let response = api_client.get_user(&token).await?;
    println!("{}", ui.apply(BOLD.apply_to(message)));
    ui::print_cli_authorized(&response.user.email, ui);
    Ok(Token { token, is_existing: true } )
}
```